rust/arrow/src/array/data.rs

```diff
@@ -219,17 +219,17 @@ impl ArrayData {
     ///
     /// Panics if `offset + length > self.len()`.
     pub fn slice(&self, offset: usize, length: usize) -> ArrayData {
-        assert!((offset + length) <= self.len());
-
-        let mut new_data = self.clone();
-
-        new_data.len = length;
-        new_data.offset = offset + self.offset;
+        self.copy_range(offset, length)
```

Comment:
This is the attempt to implement slice with MutableArrayData, but triggers ARROW-11263